katello/katello-change-hostname

```diff
@@ -50,108 +124,132 @@ def yesno
   end
 end
 
-def hammer_ping_success
-  hammer_ping = `hammer ping`
-  success = $?.success?
-  service_failed = hammer_ping.include? "FAIL"
-  success && !service_failed
+def fail_with_message(message, opt_parser=nil)
+  STDOUT.puts message
+  puts opt_parser if opt_parser
+  exit(false)
 end
 
-options = {}
-options[:program] = DEFAULT_PROGRAM
-options[:scenario] = DEFAULT_SCENARIO
-options[:system_check] = false
+def check_for_certs_tar
+  STDOUT.puts "Checking for certs tarball"
+  if @options[:certs_tar]
+    if File.file?(@options[:certs_tar])
+      true
+    else
+      fail_with_message("#{@options[:certs_tar]} does not exist! Please check the file path and try again")
+    end
+  else
+    fail_with_message("You must specify --certs-tar argument when on a #{@proxy}." \
+                      " These can be generated on the #{DEFAULT_PROGRAM} server using capsule-certs-generate and copied to this machine.")
+  end
+end
 
-opt_parser = OptionParser.new do |opt|
-  opt.banner = "Usage: katello-change-hostname HOSTNAME [OPTIONS]"
-  opt.separator  ""
-  opt.separator  "Example:"
-  opt.separator  " katello-change-hostname foo.example.com -u admin -p changeme"
-  opt.separator  ""
-  opt.separator  "Options"
+def get_fpc_answers
+  fpc_answers = YAML.load_file("/etc/foreman-installer/scenarios.d/#{FOREMAN_PROXY_CONTENT}-answers.yaml")
+  parent_fqdn = fpc_answers["foreman_proxy_content"]["parent_fqdn"]
+  register_in_foreman = true
+  base_url = fpc_answers["foreman_proxy"]["foreman_base_url"]
+  trusted_hosts = fpc_answers["foreman_proxy"]["trusted_hosts"]
+  oauth_consumer_key = fpc_answers["foreman_proxy"]["oauth_consumer_key"]
+  oauth_consumer_secret = fpc_answers["foreman_proxy"]["oauth_consumer_secret"]
+  pulp_oauth_secret = fpc_answers["foreman_proxy_content"]["pulp_oauth_secret"]
+  certs_tar = @options[:certs_tar]
+  foreman_url = fpc_answers["puppet"]["server_foreman_url"]
+
+  fpc_installer_args = " --foreman-proxy-content-parent-fqdn #{parent_fqdn}" \
+  " --foreman-proxy-register-in-foreman #{register_in_foreman}" \
+  " --foreman-proxy-foreman-base-url #{base_url}" \
+  " --foreman-proxy-oauth-consumer-key #{oauth_consumer_key}" \
+  " --foreman-proxy-oauth-consumer-secret #{oauth_consumer_secret}" \
+  " --foreman-proxy-content-pulp-oauth-secret #{pulp_oauth_secret}" \
+  " --foreman-proxy-content-certs-tar #{certs_tar}" \
+  " --puppet-server-foreman-url #{foreman_url}" \
+  " --foreman-proxy-register-in-foreman false"
```

Comment:
Why are all these needed in a hotsname change for proxies? The answers file should have the stored previous values for all of these and I would not expect them to change.